src/vw/src/main/scala/featurizer/MapFeaturizer.scala

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark.featurizer
+
+import org.apache.spark.sql.Row
+import org.vowpalwabbit.spark.VowpalWabbitMurmur
+
+import scala.collection.mutable.{ArrayBuilder}
+
+/**
+  * Featurize map of type T into native VW structure. (hash(column name + k):value)
+  * @param fieldIdx input field index.
+  * @param columnName used as feature name prefix.
+  * @param namespaceHash pre-hashed namespace.
+  * @param mask bit mask applied to final hash.
+  * @param valueFeaturizer featurizer for value type.
+  * @tparam T value type.
+  */
+class MapFeaturizer[T](override val fieldIdx: Int, val columnName: String, val namespaceHash: Int,
+                       val mask: Int, val valueFeaturizer: (T) => Double)
+  extends Featurizer(fieldIdx) {
+
+  /**
+    * Featurize a single row.
+    * @param row input row.
+    * @param indices output indices.
+    * @param values output values.
+    * @note this interface isn't very Scala-esce, but it avoids lots of allocation.
+    *       Also due to SparseVector limitations we don't support 64bit indices (e.g. indices are signed 32bit ints)
+    */
+  override def featurize(row: Row, indices: ArrayBuilder[Int], values: ArrayBuilder[Double]): Unit = {
```

Comment:
does VW expose any of their featurizers through JNI?

Reply:
so by default VW operates on input string. Starting with C# I started mapping other language types into VW features. This featurization methods are programming language specific, but I'm trying to keep them the same across languages (C#, JSON,...)